lib/HLSL/HLLowerUDT.cpp

```diff
@@ -475,3 +474,9 @@ void hlsl::ReplaceUsesForLoweredUDT(Value *V, Value *NewV) {
       CV->removeDeadConstantUsers();
   }
 }
+
+void hlsl::ReplaceUsesForLoweredUDT(Value *V, Value *NewV) {
+  ReplaceUsesForLoweredUDTImpl(V, NewV);
+  // Merge GepUse later to avoid mutate type and merge gep use at same time.
+  dxilutil::MergeGepUse(V);
```

Comment:
This is wrong.  It should be merging `NewV`.
```suggestion
  dxilutil::MergeGepUse(NewV);
```